azurelinuxagent/common/cgroupstelemetry.py

```diff
@@ -47,37 +47,37 @@ def track_cgroup(cgroup):
 
         with CGroupsTelemetry._rlock:
             if not CGroupsTelemetry.is_tracked(cgroup.path):
-                CGroupsTelemetry._tracked.append(cgroup)
+                CGroupsTelemetry._tracked[cgroup.path] = cgroup
                 logger.info("Started tracking cgroup {0}", cgroup)
 
     @staticmethod
     def is_tracked(path):
         """
         Returns true if the given item is in the list of tracked items
-        O(n) operation. But limited to few cgroup objects we have.
+        O(1) operation.
         """
         with CGroupsTelemetry._rlock:
-            for cgroup in CGroupsTelemetry._tracked:
-                if path == cgroup.path:
-                    return True
+            if path in CGroupsTelemetry._tracked:
+                return True
 
         return False
 
     @staticmethod
     def stop_tracking(cgroup):
         """
-        Stop tracking the cgroups for the given name
+        Stop tracking the cgroups for the given path
         """
         with CGroupsTelemetry._rlock:
-            CGroupsTelemetry._tracked.remove(cgroup)
-            logger.info("Stopped tracking cgroup {0}", cgroup)
+            if cgroup.path in CGroupsTelemetry._tracked:
+                CGroupsTelemetry._tracked.pop(cgroup.path)
+                logger.info("Stopped tracking cgroup {0}", cgroup)
 
     @staticmethod
     def poll_all_tracked():
         metrics = []
-
+        inactive_cgroups = []
```

Comment:
polling and start/stop tracking are done on separate threads so we need to be sure this code is thread-safe